src/uct/ib/dc/base/dc_iface.c

```diff
@@ -490,6 +490,7 @@ void uct_dc_handle_failure(uct_ib_iface_t *ib_iface, uint32_t qp_num)
         uct_dc_iface_dci_put(iface, dci);
         ucs_assert_always(ep->dci == UCT_DC_EP_NO_DCI);
     }
+    iface->super.tx.cq_available++;
```

Comment:
this is correct fix, pls add documentation - that it is because poll_cqe for mlx5 returns NULL in case of failure and the cq_avaialble is not updated for the **error cqe**